### PR TITLE
Fix for derivative wrt to part of mixed function

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR includes changes proposed by @wence- to fix an error reported in #1723 
I've added a test that fails on master and passes on this PR's branch.

### CODE REVIEW

File: tests/regression/test_split.py [hunk 1]
```diff
@@ -58,3 +58,20 @@ def test_split_function_derivative():
...
+
+    x = Function(W)
+    u, p = split(x)
+    v, q = TestFunctions(W)
+
+    I = inner(grad(u), grad(u))*dx
+    F = derivative(I, u, v)
+    J = derivative(F, u, TrialFunctions(W)[0])
+
+    assert assemble(F) and assemble(J)
```
Comment (wence-) [inline tests/regression/test_split.py:77]: Can we check that we got the right values?

File: firedrake/ufl_expr.py [hunk 2]
```diff
@@ -139,7 +139,8 @@ def derivative(form, u, du=None, coefficient_derivatives=None):
     """
     # TODO: What about Constant?
     u_is_x = isinstance(u, ufl.SpatialCoordinate)
-    if not u_is_x and len(u.split()) > 1 and set(extract_coefficients(form)) & set(u.split()):
+    uc, = extract_coefficients(u)
```
Comment (wence-) [inline firedrake/ufl_expr.py:142]: This extraction fails for coordinate derivatives `if u_is_x`, can you handle that?

**Comment by IvanYashchuk** [comment]
How to make Jenkins run the tests for the latest commit?

**Comment by wence-** [comment]
Can you just rebase on top of master and force push? Then hopefully jenkins will notice again.

**Pull request merged**